cli/cmd/plugin_uiassets_upload.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/cloudquery/cloudquery-api-go/auth"
+	"github.com/cloudquery/cloudquery/cli/internal/api"
+	"github.com/cloudquery/cloudquery/cli/internal/hub"
+	"github.com/cloudquery/cloudquery/cli/internal/publish"
+	"github.com/spf13/cobra"
+)
+
+const (
+	pluginUIAssetsUploadShort = "Upload plugin UI assets to CloudQuery Hub."
+	pluginUIAssetsUploadLong  = `Upload plugin UI assets to CloudQuery Hub.
+
+This uploads UI assets for a specific plugin version from a local build directory to CloudQuery hub.
+`
+	pluginUIAssetsUploadExample = `
+# Upload plugin UI assets to CloudQuery Hub
+cloudquery plugin uiassets upload test-team/source/test-plugin@v1.0.0`
```

Comment:
I think we don't really need a separate command for this--it should be good enough to use `cloudquery plugin publish` and check for the `--ui-dir` flag? If the concern is that it's slow to re-upload the binaries every time, maybe let's fix that instead by comparing the checksums before doing a new upload

Reply:
What if you're not publishing a plugin and simply want to fix a bug in the UI? The command architecture/setup is the same as `plugin docs upload` (which also offers `download` and `upload --sync` though)

Reply:
> What if you're not publishing a plugin and simply want to fix a bug in the UI?

I think it's simpler (and less maintenance) to have one command that does it all -- `cloudquery plugin publish` no matter what you changed. We just need to make it a bit smarter so it's not a drag to re-upload all the binaries every time.

Reply:
- publish requires a manifest.json so you'd have to create a manifest file every time there's an issue, so you'd have to add a new option to get the plugin name/version directly which would complicate the code and be confusing.
- publish requires the plugin to not be published already, so you'd have to add another flag, or maybe make it so that it skips that half of the publush flow when the ui assets flag is specified along with the plugin reference flag
- publish is a documented command so you'd have to document all this, with exceptions ("if you use this flag, this portion of the publish flow won't be run", etc)
- `plugin uiassets` is a hidden command with no downside/documentation burden, just like `plugin docs` is

Reply:
I see, thanks for the clarification. 

I was thinking that the UI publishing would be immutable after the plugin is finalized, so more like binaries, less like docs. If you fix a UI bug, you would upload a new version of the plugin, just like if you fixed a bug in the spec or config validation. I think this is similar to the Grafana model, where a UI bug will remain in a plugin until you upgrade the version.

Not versioning UI changes as part of the plugin version could lead to confusing issues down the line, if these are being cached somewhere other than our own bucket. But okay, I think we can live with it for now if it's a hidden command, meant only to be used in special circumstances.

Reply:
We're versioning the plugin UI assets separately so that there's no chance of caching issues - each UI "version" has its own uuid prefix. Grafana kept it simple but it's part of their core product, where with us the UI is completely separate from framework/plugins and only used in cloud. Because of this separate versioning needed to happen to keep things sane.

Reply:
What will the workflow be if we'd need to support some UI issue for a plugin? I've seen in support that we end up having to try a sync by checking out some tag of an older version, because the customer hasn't/can't update to the latest version.

If we had a separate UUID-based versioning scheme for the UI Framework, would it be cumbersome to see a certain version of the UI?

Reply:
I had a chat with @hermanschaaf about this to get context and I tend to agree with having separate commands to begin with until we have some real-world experience and see the tradeoffs better.